notification-config-service-api/src/main/proto/org/hypertrace/notification/config/service/v1/notification_channel.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package org.hypertrace.notification.config.service.v1;
+
+message NewNotificationChannel {
```

Comment:
These extra wrappers can go away with the mutable data pattern. Instead, update the create/update requests to take in mutable data and mutable data + id, respectively.

Reply:
I am keeping mutable data + id as a message, the create request works with mutable data

Reply:
it makes sense to have mutable + id as a message on the return side (that is, we're returning a NotificationChannel whether we get, create or update), but on the request side, we're lumping two different concerns into the same message. The `NotificationChannel` represents the full state of a channel, which today is an ID + mutable data. What happens when a second immutable field is added, say last edit date or created date? That belongs to the rule being returned, but it doesn't belong to the update API - the update API should always be taking just an identifier and the data to change.

Reply:
But the object returned from get and used in upsert is assumed to be same in ObjectStore abstraction?

Reply:
At the object store level, it works in the same objects across all APIs. From perspective of service:

Get - lookup NotificationChannel by id via store, return NotificationChannel
Update - take in id and mutable data, lookup corresponding NotificationChannel by id via store, merge changes and upsert via store. Return NotificationChannel.
Create - use mutable data to create a new NotificationChannel with generated ID and upsert via store.


So Update may be doing an extra lookup right now - but that's something we can shortcut in the implementation (just use that data to create a NotificationChannel if we have every single field and no merging is needed). By keeping that shortcut out of the API though, we can support either - a partial update where only mutable fields are changing, or a full update where we are accepting every single field. Does that make sense?